### DISCUSSION TITLE
Bot Not Working After Putting Code On Netlify Functions

### DISCUSSION THREAD

**Discussion by Y1a2s3h4** [telegraf/telegraf#1711] [Q&A]
## Below is my code 
```
const { config } = require("dotenv");
const express = require("express");
const app = express();
const cors = require("cors");
const fs = require("fs");
const { Scenes, session, Telegraf } = require("telegraf");
config({
  path: "functions\\.env",
});
const bot = new Telegraf(process.env.TELEGRAM_BOT_API);
app.use(express.json());
app.use(cors());
const superWizard = new Scenes.WizardScene(
  "super-wizard",
  (ctx) => {
    console.log(ctx.message, ctx?.callbackQuery?.message);
    ctx.reply("1) Type of message:\n1. Tech\n2. Non-Tech");
    ctx.session.command = ctx.message.text;
    return ctx.wizard.next();
  },
  ...............
  ...............
  // here my all the other wizards scene code is
});

// superWizard actions, bot commands, and other stuff code.
const stage = new Scenes.Stage([superWizard]);
bot.catch((err, ctx) => {
  console.log(`Ooops, encountered an error for ${ctx.updateType}`, err);
});
bot.use(session());
bot.use(stage.middleware());

bot.launch();

exports.handler = async (event) => {
  try {
    return { statusCode: 200, body: "Done!" };
  } catch (e) {
    console.error("error in handler:", e);
    return {
      statusCode: 400,
      body: "This endpoint is meant for bot and telegram communication",
    };
  }
};
````

## Bot not giving any response.
![image](https://user-images.githubusercontent.com/45822567/193880057-941bfb60-4a8e-402a-829d-afe3e39887ac.png)

**Comment by MKRhere** [comment]
Remove `bot.launch()` for webhook bots. Instead, in your handler function, do `bot.handleUpdate(JSON.parse(event.body))`. Going by https://docs.netlify.com/functions/build/?fn-language=ts.

You still need a way to tell Telegram about your bot's webhook URL. After deploying, get the full URL to the function, and do this from inside your project (only needed once, until you function URL changes):

```shell
npx telegraf -t $BOT_TOKEN -m setWebhook -D '{ "url": $FULL_URL_TO_FUNCTION }'
```

Replace the variables with the correct values.

**Reply by Y1a2s3h4** [reply]
### It worked but bot stops sometimes and your code gives an error
![image](https://user-images.githubusercontent.com/45822567/194005541-fabee457-63d6-4fb8-b7c8-ecdd98064b33.png)

**Reply by Y1a2s3h4** [reply]
### And this is the logs from netlify functions.
![image](https://user-images.githubusercontent.com/45822567/194006186-67451ea0-fb57-4b55-a0ce-3706b7ca9924.png)
